File: LeetCode/Easy/0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        cur = head
        while cur and cur.val == val:
            cur = cur.next
            head = cur

        while cur:
            if cur.next and cur.next.val == val:
                cur.next = cur.next.next
            else:
                cur = cur.next

        return head

File: LeetCode/Easy/0203-remove-linked-list-elements/0203-remove-linked-list-elements.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    cur := head
    for cur != nil && cur.Val == val {
        cur = cur.Next
        head = cur
    }
    
    for cur != nil {
        if cur.Next != nil && cur.Next.Val == val {
            cur.Next = cur.Next.Next
        } else {
            cur = cur.Next
        }
    }
    return head
}